src/software/simulated_tests/terminating_validation_functions/robot_blocking_shot.cpp

```diff
@@ -0,0 +1,39 @@
+#include "software/simulated_tests/terminating_validation_functions/robot_blocking_shot.h"
+
+#include "software/logger/logger.h"
+
+
+void robotBlockingShot(RobotId robot_id, std::shared_ptr<World> world_ptr,
```

Comment:
would it be sufficient to make a triangle from the two goal posts and the ball and check if the robot position is contained in that triangle? (i.e. check `contains(triangle, robot.position())`)

Reply:
Oh that is a good point, I didn't think of that solution. There's already a validation function called robot_in_polygon so I can reuse that validation function to check if our friendly robot is in the triangle.